### PR TITLE
Fix custom http client not being used when fetching related resources

### PR DESCRIPTION
**Pull request by forivall**
I found this issue when I was first trying to fix strictSSL issues when fetching the WSDL, but initially worked around it with `wsdl_options`. However, since I will need to rewrite some urls on the fly, I need the full custom client to work when fetching the wsdl and all related resources.

`_socketStream.js` can also be used to clean up `client-customHttp-test`, on which I based my test (which is why I put it in a separate file).

In summary: a 1 line fix, with 208 additional lines of test code. :beers:

### CODE REVIEW

File: lib/wsdl.js [hunk 1]
```diff
@@ -1124,6 +1124,7 @@ WSDL.prototype._initializeOptions = function (options) {
   // Allow any request headers to keep passing through
   this.options.wsdl_headers = options.wsdl_headers;
   this.options.wsdl_options = options.wsdl_options;
+  if (options.httpClient) this.options.httpClient = options.httpClient;
```
Comment (herom) [inline lib/wsdl.js]: Could you please either use the "curly brace" form of the `if` condition? Personally I think it makes the code more readable and I hope that I get a `jscs` rule set in `soap` as soon as possible :+1:

**Comment by forivall** [comment]
@herom updated! My personal preference is braces when the body of the statement is on the next line, but I'm fine with following your style. JSCS / eslint would be great in communicating your preferences :+1:

**Comment by herom** [comment]
Thanks a lot for your time & effort @forivall - we appreciate it a lot :+1:
